soweego/importer/importer.py

```diff
@@ -47,32 +49,42 @@ def import_cli(catalog: str, resolve: bool, output: str) -> None:
     importer.refresh_dump(output, extractor, resolve)
 
 
+def _resolve_url(res):
+    return url_utils.resolve(res.url), res
+
+
 @click.command()
 @click.argument('catalog', type=click.Choice(target_database.available_targets()))
 def validate_links_cli(catalog: str):
     for entity_type in target_database.available_types_for_target(catalog):
 
         LOGGER.info("Validating %s %s links..." % (catalog, entity_type))
         entity = target_database.get_link_entity(catalog, entity_type)
+        if not entity:
+            LOGGER.info("%s %s does not have a links table. Skipping..." % (catalog, entity_type))
+            continue
 
         session = DBManager.connect_to_db()
-        total = 0
+        total = session.query(entity).count()
         removed = 0
 
-        # Validate each link
-        for res in session.query(entity):
-            total += 1
-            if not url_utils.resolve(res.url):
-                # if not valid delete
-                session.delete(res)
-                session.commit()
-                removed += 1
+        with Pool() as pool:
+            # Validate each link
+            for resolved, res_entity in tqdm(pool.imap_unordered(_resolve_url,
+                                                                 session.query(entity)), total=total):
+                if not resolved:
+                    session_delete = DBManager.connect_to_db()
+                    # if not valid delete
+                    session_delete.delete(res_entity)
```

Comment:
Perhaps we should surround the following lines with the usual `try session.commit / except session.rollback / finally session.close` ?

Reply:
It is a good idea. I'm going to fix it right now!